pkg/virt-api/rest/subresource.go

```diff
@@ -626,8 +627,24 @@ func (app *SubresourceAPIApp) StopVMRequestHandler(request *restful.Request, res
 			writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf(vmNotRunning)), response)
 			return
 		}
-		writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("%v does not support manual stop requests", v1.RunStrategyHalted)), response)
-		return
+		if bodyStruct.GracePeriod == nil {
+			writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("%v only supports manual stop requests with graceperiod=0", v1.RunStrategyHalted)), response)
+			return
+		}
+		if *bodyStruct.GracePeriod >= *vmi.Spec.TerminationGracePeriodSeconds {
+			writeError(errors.NewConflict(v1.Resource("virtualmachine"), name, fmt.Errorf("%v only supports manual stop requests with graceperiod=0", v1.RunStrategyHalted)), response)
```

Comment:
Can you adjust the error message to point out that only values less then the current graceperiod are accepted. please? Here it seems that only 0 is a valid value. Thanks